### ISSUE TITLE
driver / AMI430 handles invalid values incorrectly

### ISSUE THREAD

**Issue by WilliamHPNielsen** [QCoDeS/Qcodes#1143]
The AMI430 driver does not handle invalid target field values correctly. I could only test this with the simulated instrument, but as far as I can see from the source, the real instrument will have the same behaviour.

### Steps to reproduce

```python
import numpy as np

import qcodes.instrument.sims as sims
from qcodes.instrument_drivers.american_magnetics.AMI430 import AMI430_3D, AMI430Warning
from qcodes.instrument.ip_to_visa import AMI430_VISA
from qcodes.math.field_vector import FieldVector

field_limit = [
    lambda x, y, z: x == 0 and y == 0 and z < 3,
    lambda x, y, z: np.linalg.norm([x, y, z]) < 2
]

# path to the .yaml file containing the simulated instrument
visalib = sims.__file__.replace('__init__.py', 'AMI430.yaml@sim')


mag_x = AMI430_VISA('x', address='GPIB::1::65535::INSTR', visalib=visalib,
                    terminator='\n', port=1)
mag_y = AMI430_VISA('y', address='GPIB::2::65535::INSTR', visalib=visalib,
                    terminator='\n', port=1)
mag_z = AMI430_VISA('z', address='GPIB::3::65535::INSTR', visalib=visalib,
                    terminator='\n', port=1)

driver = AMI430_3D("AMI430-3D", mag_x, mag_y, mag_z, field_limit)

#######################################################################
# Example starts here

print(f'x is set to {driver.x()}')
try:
    invalid_value = 999
    print(f'Trying to set x to {invalid_value}')
    driver.x(invalid_value)
except ValueError:
    print('Did NOT set x')
print(f'x is set to {driver.x()}')
```

### Expected behaviour
The set point should not update when the value is invalid.

### Actual behaviour
The set point *does* update.

@QCoDeS/core

**Comment by WilliamHPNielsen** [comment]
@sohailc, perhaps it is easiest for you, as the original author, to look into this one?

**Comment by sohailc** [comment]
@WilliamHPNielsen I am on it

**Comment by sohailc** [comment]
https://github.com/sohailc/Qcodes/tree/fix/ami_invalid_field_limits_set

**Comment by sohailc** [comment]
https://github.com/QCoDeS/Qcodes/pull/1162

**Comment by jenshnielsen** [comment]
Fixed by #1162
